### ISSUE TITLE
Continuity and Mojave

### ISSUE THREAD

**Issue by PeterHolbrook** [dokterdok/Continuity-Activation-Tool#438]
Has anyone verified if machines theoretically non-Continuity compliant (unless they have the proper Bluetooth/Wi-Fi card) will continue to work in macOS 10.14 Mojave? Will the new Handoff/Continuity features work?

**Comment by iMonZ** [comment]
I think not because 10.13.3 is the last version that worked

**Comment by PeterHolbrook** [comment]
Why do you say 10.13.3 is the last version that worked? I'm on 10.13.5 on a Mac Pro 5,1  and Continuity works just fine.

**Comment by finbarrcnaipe** [comment]
@PeterHolbrook :  Yep!  Me too, MacBook Pro Early 2011 w/updated CAX BTLE card...works great!   Have not tried Mojave yet, and don’t plan to until at least official release.  (Plus I’m annoyed anyway that Apple is dumping support for most all of my Macs except for my Mac Mini 2012.   Using the open source patch tool,  I’m running High Sierra on a MacBook Pro 2009 Unibody and it works perfectly without issue.   So dropping support for older machines is complete BS in MANY cases.  Same will go for Mojave...there is no technical reason it cannot run on an iMac 2010 w/Quad-core i7 and 32MB of RAM.    Apple continues to piss off their original enthusiasts and evangelists...   It is a shame what has happened to them.

**Comment by airmanchairman** [comment]
Aye, me too with the exact same hardware and macOS configuration as DOCSISMAN (early 2011 17” unibody MBP  + updated 4CAX card running High Sierra 11.13.5), and Continuity / Handoff / etc works like a boss!

**Comment by iMonZ** [comment]
I got with 10.13.5 a bootloop when I try to install it

**Comment by PeterHolbrook** [comment]
You should investigate why it doesn't work for you. It's normally flawless.

**Comment by gt2416** [comment]
I have a MacBook Pro 2011 which is NOT supported by mojave. I have an upgraded internal wifi card.
Using dosdude1 tool I was able to install Mojave and install the continuity tool.
Whats broken:
- Can no longer initiate hotspot from phone. It fails to activate every time.
- Insert from iPad/iPhone aka continuity camera does not work. Error says "Take Photos" service cannot be used. Although it looks to try and can see my device.
- Facetime app fails to initiate calls from iPhone unless its a facetime call. Error says make sure iphone is using same iCloud account. 

Everything else seems to work. Can continue from safari, messages work, shared clipboards also works.
It would be great to get a proper update for Mojave as its literally the only significant upgrade to macOS in years.

**Comment by airmanchairman** [comment]
@gt2416 - Awesome! 👏🏽👏🏽

**Comment by gt2416** [comment]
Just some updates
- Continuity camera suddenly started working.  (Did not reboot or anything) Tested with an iPad.
- Not being able to call phone numbers MAYBE because my iPhone is not running ios 12 and I will not update it till the official release.

Any body else want to try it with Mojave ? The public beta is actually pretty stable. I've switched to it completely on my mac.

**Comment by airmanchairman** [comment]
@gt2416 what is this “Camera Continuity” feature that’s completely missed me by?

**Comment by PeterHolbrook** [comment]
If I'm not mistaken, a Mac using "Camera Continuity" is capable of "seeing" and capturing what its paired iPhone "sees".

**Comment by PeterHolbrook** [comment]
@gt2416: It's a long time since I last checked this out, but I think that, back when I updated to Sierra 10.12, my hotspot wouldn't work even after running CAT. I had to manually edit one or two SIP-protected files for it to work. I think, somewhere in this forum, I mentioned the steps I undertook. I am almost certain that if you follow my old instructions, you'll succeed.

**Comment by gt2416** [comment]
Could you point be to that info if you have it please ? I've searched for a while but can't find it.

**Comment by PeterHolbrook** [comment]
No, sorry. I would have to look for it myself. I guess it will be among CAT issues relative to Sierra or High Sierra.

EDIT: I found it! Have a look at what I said in https://github.com/dokterdok/Continuity-Activation-Tool/issues/397 and the links provided therein. Good luck.

**Comment by gt2416** [comment]
Thank you SO much for the link ! 
I actually did that already, it was part of my install notes. Guess I had problems on High Sierra or something so I had to do it. Thanks for the suggestion tho !
Airdrop and stuff works, its really only those specific things I mentioned that dont work. Continuity definitely works and functions well except for those things. 
The personal hotpot cant start but it can see my phone.

**Comment by fabiojuniorrr** [comment]
I have a MacBook Pro 5,5 running mojave through dosdude1 patch tool. I wonder if cat will work, I got a asus bluetooth 4.0 dongle. Thanks!

**Comment by MDz1985** [comment]
I have macbook air 4.2. Only Airdrop works for me on Mojave (On the last build of High Sierra all works well). Is there anybody who have old Handoff working?

**Comment by gejobj** [comment]
I have a MacMini5,1 from 2011 which is NOT supported by mojave and using dosdude1 tool everything works perfect (except gray transparency glitch). I solved it with: "System Preferences" -> “Accessibility” -> “Display" -> select “Reduce transparency”.

As @gt2416 told before, there are a few continuity features don't work but all other things works perfect such as Airdrop or app switch.

**Comment by MDz1985** [comment]
App switch doesn't work for me (if you meant that in the picture below)... O'k. Will try to fix. Thanks.

![](https://help.apple.com/assets/5A8F665C094622DB1E5375F8/5A8F665D094622DB1E5375FF/en_US/b5836874e5ec71781b9d5fc6f458dd44.png)

**Comment by gejobj** [comment]
I can see iMessages and Mail from my iphone in my macmini.
![captura de pantalla 2018-10-04 a las 19 51 45](https://user-images.githubusercontent.com/9005888/46492871-1b20d000-c80f-11e8-8f04-880e3f52b578.png)
![captura de pantalla 2018-10-04 a las 19 52 01](https://user-images.githubusercontent.com/9005888/46492872-1b20d000-c80f-11e8-9f4d-185fa126919a.png)

**Comment by maciboy** [comment]
Hi :)
So, if I understand the posts here correctly, it works to patch Continuity on a patched (dosdude1 tool) Mojave Mac?
I thought there might be issues with the bluetooth kexts as the dosdude1 Mojave patcher might altered the kexts and CAT will alter them again, breaking Mojave ^^
If that’s not the case and sb can again confirm, I would love to patch my patched Mojave :D

**Comment by gejobj** [comment]
> Hi :)
> So, if I understand the posts here correctly, it works to patch Continuity on a patched (dosdude1 tool) Mojave Mac?
> I thought there might be issues with the bluetooth kexts as the dosdude1 Mojave patcher might altered the kexts and CAT will alter them again, breaking Mojave ^^
> If that’s not the case and sb can again confirm, I would love to patch my patched Mojave :D

Yes, if you patch bluetooth again using dosdude1 patch app, your dosdude1-CAT patched bluetooth file will be broken and you should reinstall mojave.
I did that on Saturday and I have to reinstall all again T_T
Maybe dosdude should add CAT into his patched files to avoid this.
Regards.

**Comment by maciboy** [comment]
Thanks!
Uhh that‘s bad luck:/
In the Macrumors thread, I already suggested to add a CAT patch to the list of available patches, but I received no answer...

BTW, do you know if one can disable the Dosdude1 patcher updates?

**Comment by gejobj** [comment]
Wow! that's sounds bad.
I don't know if I can disable patcher updates or a specific patch. The problem will be when apple release 10.14.1 version of mojave and we have to apply all patch again.
I hope dosdude1 read your post at macrumors and add it to the list.
Regards.

**Comment by ovrdrvn** [comment]
I get this error now with CAT:

Verifying Wi-Fi hardware...             WARNING. An unknown/untested AirPort card using the 'BrcmNIC' kext is active. This tool can't fix this.

**Comment by grzegorz914** [comment]
Same error for me. 

> Verifying Wi-Fi hardware... WARNING. An unknown/untested AirPort card using the 'BrcmNIC' kext is active. This tool can't fix this.

iMac 11.3, Mojave, Apple Broadcom BCM94360CD, GTX 765M
AirPort Extreme  (0x14E4, 0x111)
Broadcom BCM43xx 1.0 (7.77.61.1 AirPortDriverBrcmNIC-1305.2)

**Comment by glaucorossi** [comment]
Working well on Mojave 10.14 (MacBook Pro A1278 Late 2011)

**Comment by ghost** [comment]
Working well on Mojave 10.14 (MacBook Pro A1278 Late 2011)

**Comment by GilDev** [comment]
Got a MacBook Pro Early 2011 (MacBookPro8,1) running patched Mojave 10.14.1.
I think I am going to order a BCM94331PCIEBT4CAX card to make everything work with my iPad Pro.
Can someone please post instructions as to how to make everything work with the patched Mojave installation? It seems @glaurossi and @gt2416 got it working with similar MacBook Pro?

**Comment by gt2416** [comment]
@GilDev 
- I would 100% recommend getting the BCM94331PCIEBT4CAX card. Honestly I have no reason to 
  buy a new Mac anymore. Really only wifi AC is missing. 
- For the MacBookPro8,1 I believe the correct board number is "94245B3640C91C81" There are different models for 8,1 on Apple's website so not sure if yours is the same as mine. Either way it wont hurt even if its wrong. 

Step 1: Log out of iCloud
Step 2: Use the latest beta continuity Patcher, restart.
Step 3: Edit the file `/System/Library/Frameworks/IOBluetooth.framework/Versions/A/Resources/SystemParameters.plist`
​Look for the board number "94245B3640C91C81" and change the continuity flag to "true".
I did this through nano on terminal to make dealing with permissions easier.
Step 4: Run this command in terminal
`sudo -E perl -pi -e "s/\Mac-00BE6ED71E35EB86/\Mac-94245B3640C91C81/" /System/Library/Extensions/IO80211Family.kext/Contents/PlugIns/AirPortBrcm4360.kext/Contents/MacOS/AirPortBrcm4360`
Step 4: Restart
Step 5: Log into iCloud, restart. Test 
​

**Comment by GilDev** [comment]
Thanks a lot @gt2416! I have one reason to buy a new MacBook Pro, it would be nice if it had a better CPU. =) But I’ve just built an Hackintosh so this part is fine for now.
I’ve found [this card](https://fr.aliexpress.com/item/BCM94331PCIEBT4CAX-Bluetooth-4-0-WiFi-sans-fil-Carte-AirPort-Pour-MacBook-Pro-13-3-MD101LL-UN/32913635901.html) for 20 € so I think I’ll order it.

**Comment by grzegorz914** [comment]
> @GilDev
> 
> * I would 100% recommend getting the BCM94331PCIEBT4CAX card. Honestly I have no reason to
>   buy a new Mac anymore. Really only wifi AC is missing.
> * For the MacBookPro8,1 I believe the correct board number is "94245B3640C91C81" There are different models for 8,1 on Apple's website so not sure if yours is the same as mine. Either way it wont hurt even if its wrong.
> 
> Step 3: Edit the file `/System/Library/Frameworks/IOBluetooth.framework/Versions/A/Resources/SystemParameters.plist`
> ​Look for the board number "94245B3640C91C81" and change the continuity flag to "true".
> I did this through nano on terminal to make dealing with permissions easier.
> Step 4: Run this command in terminal
> `sudo -E perl -pi -e "s/\Mac-00BE6ED71E35EB86/\Mac-94245B3640C91C81/" /System/Library/Extensions/IO80211Family.kext/Contents/PlugIns/AirPortBrcm4360.kext/Contents/MacOS/AirPortBrcm4360`
> Step 4: Restart
> Step 5: Log into iCloud, restart. Test
> ​

This method work like a charm on iMac mid 2010 Mojave and replaced AirPort card to Apple Broadcom BCM94360CD. 
No need to log out from iCloud.
Also unlock with Apple Watch working.

In terminal get Your board number:
`ioreg -l | grep "board-id" | awk -F\" '{print $4}'`

My board number: Mac-F2238BAE - set flag to "true" in:
`sudo /usr/libexec/PlistBuddy -c "Set:Mac-F2238BAE:ContinuitySupport true" "/System/Library/Frameworks/IOBluetooth.framework/Versions/A/Resources/SystemParameters.plist"`

then do this 2 command in Terminal:
```
sudo -E perl -pi -e "s/\Mac-00BE6ED71E35EB86/\Mac-F2238BAE/" /System/Library/Extensions/IO80211Family.kext/Contents/PlugIns/AirPortBrcm4360.kext/Contents/MacOS/AirPortBrcm4360

sudo -E perl -pi -e "s/\Mac-00BE6ED71E35EB86/\Mac-F2238BAE/" /System/Library/Extensions/IO80211Family.kext/Contents/PlugIns/AirPortBrcmNIC.kext/Contents/MacOS/AirPortBrcmNIC
```

Restart and all working OK.

Thanks for the tip.

**Comment by airmanchairman** [comment]
Mojave on early 2011 MacBook Pro? Did I hear y'all correctly?

How?

**Comment by GilDev** [comment]
@airmanchairman Just use [dosdude1's patcher](http://dosdude1.com/mojave). It's easy and works great, just use dark theme (which I love) and maybe disable “Transparency effects” in Accessibility settings to avoid graphical glitches. Other than that, no differences from my Hackintosh!

**Comment by airmanchairman** [comment]
Wow! I must’ve been living under a rock all this time! 😎

Thanks 🙏🏾

**Comment by erujp** [comment]
> This method work like a charm on iMac mid 2010 Mojave and replaced AirPort card to Apple Broadcom BCM94360CD.

@grzegorz914
This method worked for me on MacBookAir 4,1 (mid 2011) in Mojave 10.14, even if the copy-paste function was too slow or absent, but it doesn't work anymore in Mojave 10.14.1. I've repeated the whole process (from CAT beta to everything suggested by GilDev and grzegorz914), with no result.
Any suggestion?

**Comment by grzegorz914** [comment]
Im updated Mojave from 10.14.1 to 10.14.2b1 and all working OK.
I do not use CAT, I use only method what I previously described.

**Comment by erujp** [comment]
> Im updated Mojave from 10.14.1 to 10.14.2b1 and all working OK.
> I do not use CAT, I use only method what I previously described.

Your reply to @GilDev began with "This method work like a charm", and since @GilDev's method at step 2 reads "Step 2: Use the latest beta continuity Patcher", I thought that you too used CAT. 
Anyway, now I don't know what to do…

Just for curiosity: why do the terminal commands use "Mac-00BE6ED71E35EB86" in the command line? That's the board-id for iMac 13,1.

**Comment by grzegorz914** [comment]
OK, this was misunderstood, I start from step 3 because BCM94360CD is nativ supported under Mojave, no need to patch original *.kext. I need only add my Board Number as supported to the original *.kext.

The terminal command overwrite first supported Board Number with Yours Board Number. That’s all.

**Comment by erujp** [comment]
> I start from step 3 because BCM94360CD is nativ supported under Mojave

I've re-installed Mojave 10.14.1 and done as you said, but it doesn't work. In System Information it says that Handoff is not supported and in System Preferences/General there is no checkbox for Handoff.

If I use CAT, System Information says that Handoff is supported and in System Preferences/General appears the checkbox for Handoff. In 10.14, after I used CAT and applied the changes suggested by @GilDev, Handoff and Continuity worked. In 10.14.1 I did the same exact thing, but it doesn't work. 
Well, I'll wait for the next update!

**Comment by grzegorz914** [comment]
@erujp
Are U set the flag to “true” for Your Board Number.?

**Comment by erujp** [comment]
> @erujp
> Are U set the flag to “true” for Your Board Number.?

Yes, that's the only thing that made it work in 10.14. But this time it doesn't work. Anyway, that's not the first time that Continuity and Handoff work this way. In Sierra it worked fine, in High Sierra it didn't work. In Mojave 14.0 worked, and now it doesn't. I think there's a combination of different little things, that's why everyone has different results.

**Comment by ovrdrvn** [comment]
@gt2416 For a Mac Pro 5.1 with a compatible BT/WIFI board that worked pre Mojave, would you know how to edit the file and run the command you mentioned?

**Comment by maciboy** [comment]
MacBook Pro 6,2 (Mid 2010, 15“) with external BT 4.0 USB dongle:
**Mojave 10.14.1**

During patch process:
![bildschirmfoto 2018-11-07 um 21 45 56](https://user-images.githubusercontent.com/19333006/48162522-9f1e1a00-e2dd-11e8-97d9-b347e1298c8e.png)

After patching and reboot:
![bildschirmfoto 2018-11-07 um 21 50 05](https://user-images.githubusercontent.com/19333006/48162551-b2c98080-e2dd-11e8-9677-648bfd6863b5.png)

AirDrop is working 👍 
(No kernel panics or else…)

**Comment by ovrdrvn** [comment]
@gt2416 YOU ROCK! I followed your instructions closely and it does indeed work. This is a huge relief and I commend you for helping the community. Maybe a clean set of instructions should be posted for all those challenged.

**Comment by MDz1985** [comment]
With CAT only Airdrop works for me on Mojave (MacBook Air 4.2). With instructions of **grzegorz914** handoff works well!!! ALL WORKS WELL (even "Get photo from iPhone/iPad"). THANK YOU!!!!!

**Comment by GilDev** [comment]
@gt2416 @grzegorz914 Okay guys, I’ve just changed my WLAN card on my MacBookPro8,1 for the BCM94331PCIEBT4CAX I just received. I booted and WiFi and Bluetooth were working out of the box. System Informations (maybe it has another name in English?) says “Low-Energy Bluetooh: Yes” so everything seems to be in order.
Sadly, I’ve followed @grzegorz914’s steps but using my own board number which was indeed the one given by @gt2416. Still no continuity, handoff or AirDrop. System Informations says Handoff is not supported. I tried logging out of iCloud then in again but nothing. To be clear, I haven’t used CAT but as WiFi & Bluetooth are working OOTB, maybe it’s not useful?

**Comment by Zayg94** [comment]
@grzegorz914 

Glad I found this! I've been doing my best to modify my 13" MacBook Pro 8,1 (Early 2011) to continue lasting as long as I can.
I've used CAT before and never fully worked for me. 
Updated my Mac to run Mojave using DosDude1's patcher and that turned out flawlessly (Running 10.14.1). 
Followed all your steps. This is what I did for Step 3 (After so much searching for an easy to use .plist editor): Downloaded BBEdit and made the changes there. 
Tested and found that the main continuity features I use worked.
Airdrop - Works
Handoff - Works

Thanks!

**Comment by GilDev** [comment]
@Zayg94 It’s weird it doesn’t work with mine. I’ll try later with your plist editor. I only used a text editor, but it should still work.

**Comment by Zayg94** [comment]
@GilDev I initially tried with TextEdit but it didn't allow me to save any changes. BBEdit is the only one that let me. 

Update:
Universal Clipboard - Works
Instant Hotspot - Works
Continuity Camera - Works
Auto Unlock - Apparently I need to have an AC wifi module. Seems there isn't one for the Early 2011 so I won't be able to enjoy that feature.

**Comment by GilDev** [comment]
Well the file has definitely been updated, but still no AirDrop nor everything.
Why won’t it work for me?!

**Comment by MDz1985** [comment]
@GilDev Do you use correct  board number?
Check:
ioreg -l | grep "board-id" | awk -F\" '{print $4}'
 
Ps: I also used TextEdit (previously added permissions for folder and file)  
May be[ it ](
https://forums.macrumors.com/threads/handoff-breakdown-with-activation-tool.1803192/#post-20124161)will help you

**Comment by GilDev** [comment]
Yes it is the correct board number.
I have used CAT yesterday in order to patch my MBP, and although I got [a bunch of errors](https://pastebin.com/0m7nUdnt) when installing the patch, I can now activate Handoff in System Settings and AirDrop works. But, Handoff, Continuity and everything still doesn’t work, only AirDrop…
Now I tried uninstalling the patch from CAT app, in order to try to solve the errors it gave me, but it wont let me uninstall, instead it gives me this:
```--- Initiating uninstallation ---

Restoring kexts...                      OK. No backups made with the tool were found, using the OS X Recovery Disk backups.
Mounting Recovery HD...                 OK
Reinstalling original Apple drivers...  NOT OK
Applying correct permissions...         OK
Updating kext caches...                 OK
Updating system caches...               OK
Disabling any HCI switch behavior...    OK
Verifying system kexts signatures...    OK. 1 or more unsigned drivers were found. OS kext security protection won't be changed to prevent issues.
Patching ContinuitySupport...           OK.
2018-11-22 01:04:00.423 osascript[2380:18442] isPrefsCreateCacheFromEnabledAndDefaultInputSources - can't find anything from GetInputSourceEnabledPrefs, use defaultASCIIKeyLayoutDict = <CFBasicHash 0x7fb3c8417430 [0x7fff9a1a98f0]>{type = mutable dict, count = 3,
entries =>
	0 : <CFString 0x7fff9a217eb8 [0x7fff9a1a98f0]>{contents = "InputSourceKind"} = <CFString 0x7fff9a25d3f8 [0x7fff9a1a98f0]>{contents = "Keyboard Layout"}
	1 : <CFString 0x7fff9a249538 [0x7fff9a1a98f0]>{contents = "KeyboardLayout ID"} = <CFNumber 0x812e17c757bd6b11 [0x7fff9a1a98f0]>{value = +1, type = kCFNumberSInt64Type}
	9 : <CFString 0x7fff9a212bb8 [0x7fff9a1a98f0]>{contents = "KeyboardLayout Name"} = French
}
OK. Automatic continuity check disabled.

DONE. Please reboot now to complete the uninstallation.
```

And the patch stays activated…

**Comment by gt2416** [comment]
@GilDev Not sure how to fix your current problem except to start from scratch again and maybe try a pram reset ? You should do that after swapping hardware.

**Comment by bluefoxlee** [comment]
MacBook 4,1 with 10.14.1. Big thanks @gt2416 and @grzegorz914 method works for me. wow!.

Not need to log out iCloud. however, it did not work after first reboot. turn off the "Handoff" and turn back on did the work. Thanks.

Updated: Even works great on one of my hackintosh.

**Comment by ryoplasmic** [comment]
> > @erujp
> > Are U set the flag to “true” for Your Board Number.?
> 
> Yes, that's the only thing that made it work in 10.14. But this time it doesn't work. Anyway, that's not the first time that Continuity and Handoff work this way. In Sierra it worked fine, in High Sierra it didn't work. In Mojave 14.0 worked, and now it doesn't. I think there's a combination of different little things, that's why everyone has different results.

Just want to let you know my MBA4,1 works on 10.14.1 and 10.14.2

When I upgrade to 10.14.2, the system messed up a bit and it doesnt work and even see the no wifi hardware issues; but I uninstall CAT inside the CAT terminal, reboot, reapply, and it works again. 

It works even the SystemParameters.plist entry is still false. Previously I have to modify it.

Hope this helps

**Comment by codethought** [comment]
@grzegorz914
 
> This method work like a charm on iMac mid 2010 Mojave and replaced AirPort card to Apple Broadcom BCM94360CD.
> No need to log out from iCloud.
> Also unlock with Apple Watch working.
> 
> In terminal get Your board number:
> `ioreg -l | grep "board-id" | awk -F\" '{print $4}'`
> 
> My board number: Mac-F2238BAE - set flag to "true" in:
> `/System/Library/Frameworks/IOBluetooth.framework/Versions/A/Resources/SystemParameters.plist`
> 
> then do this 2 command in Terminal:
> 
> ```
> sudo -E perl -pi -e "s/\Mac-00BE6ED71E35EB86/\Mac-F2238BAE/" /System/Library/Extensions/IO80211Family.kext/Contents/PlugIns/AirPortBrcm4360.kext/Contents/MacOS/AirPortBrcm4360
> 
> sudo -E perl -pi -e "s/\Mac-00BE6ED71E35EB86/\Mac-F2238BAE/" /System/Library/Extensions/IO80211Family.kext/Contents/PlugIns/AirPortBrcmNIC.kext/Contents/MacOS/AirPortBrcmNIC
> ```
> Restart and all working OK.
> 
> Thanks for the tip.

THIS.. THIS worked on my 2012 cMP 5,1.  I just used the machine ID from the ioreg command..  Mac-F221BEC8  and swapped accordingly. The ONE other thing I had to do was turn Handoff off and back on in system preferences.  Thanks!

**Comment by chrisdata** [comment]
my board no shown: Mac-F22587C8
Will the method above work for my mac?

**Comment by grzegorz914** [comment]
U need Use Your Board ID

**Comment by chrisdata** [comment]
My mac is MBP5,3 2009 and no change to new airport board, will the method above still work?

**Comment by kertenkel** [comment]
Registered to tell you this works really nice...!!
(had to reinstall mojave 3 times to confirm.)

dosguy1 patched (thanks!) 
- Mojave 10.14.2 on 15" 2011-late MbP.
- Iphone 7 / ios 12: 
Airdrop > 👍 
Continuity > 👍 
+
**Cheap BT4 dongle.** 
(You need extra BT4 dongle for this to work properly, CAT will detect it and patch accordingly.)

**Steps:**
--------------
1) Apply CAT, restart from within tool.
Note: during first reboot screen will go black, **don't manually restart!!!** Your wifi drivers will go corrupt if you do that)
--------------
2) Get BBEdit trial.
3) Finder > Go to Folder > /System/Library/Frameworks/IOBluetooth.framework/Versions/A/Resources/
4) Right click "SystemParameters.plist" > Edit with BBedit.
5) CMD+ Space: terminal > ioreg -l | grep "board-id" | awk -F\" '{print $4}'
**Extra)** Confirm board-id from here: https://github.com/upekkha/AppleHardwareTest
6) Copy Paste from terminal > "**Mac-urmotherboardid**"
7) Go back to "SystemParameters.plist" > Search for "**Mac-urmotherboardid**" > Set false\ flag to true\
8) Go back to terminal > 
`sudo -E perl -pi -e "s/\Mac-00BE6ED71E35EB86/\**Mac-urmotherboardid**/" /System/Library/Extensions/IO80211Family.kext/Contents/PlugIns/AirPortBrcm4360.kext/Contents/MacOS/AirPortBrcm4360`

`sudo -E perl -pi -e "s/\Mac-00BE6ED71E35EB86/\**Mac-urmotherboardid**/" /System/Library/Extensions/IO80211Family.kext/Contents/PlugIns/AirPortBrcmNIC.kext/Contents/MacOS/AirPortBrcmNIC`

9) Log out of icloud & restart.
10) System Preferences > Allow Handoff (tick / untick several times)
11) Login to icloud.
12) Right click desktop > Import from Iphone should be working now.

Thank you all for making this work 💯 👍 👍 👍

**Comment by timetoseek** [comment]
There's an alternate way which worked for me: first install High Sierra, use the CAT tool, then upgrade to Mojave...  Please see steps here:  https://github.com/dokterdok/Continuity-Activation-Tool/issues/450#issuecomment-453380542

**Comment by GilDev** [comment]
Tried again this evening with @kertenkel and @gt2416 instructions on my MacBook Pro 8,1 early 2011, but now with the 10.14.2 update. Still got tons of errors from CAT but it worked!
Continuity, Handoff and AirDrop (which was the only thing working on 10.14.1) all work, I can now take photos from my iPad Pro without any problems. Thanks everyone!

**Comment by Mrt0t0** [comment]
Macbook Pro Early 2011 8.1
Toshiba SSD master 256Go
Switch optical drive with a HDD hybrid Seagate 1T
16g ddr3 1333mhz
BT4.0 from 2012 MBP
10.14.3 working very good ;) Thks @dosdude1 @GilDev @gt2416 

Tips:
1) Update your MB firmware to lastest
2) Clean install 10.14.3 + patch dosdude1 /or/ clean 10.14.1 + OTA 10.14.3 + patch update dosdude1
3) Active Trim SSD (sudo trimforce enable) check if unable in Apple Icon/Syst Info/storage.
3) Patch BT 4.0 (use @GilDev and @gt2416 method) 

Still working with Mojave 10.14.3 on a MBP early 2011 8.1

https://github.com/dokterdok/Continuity-Activation-Tool/issues/438

https://forums.macrumors.com/threads/macos-10-14-mojave-on-unsupported-macs-thread.2121473/

**Comment by tedteng85** [comment]
Early 2011 MBP, BCM94331PCIEBT4CAX, CAT worked with High Sierra. 
Used dosdude1's patch updater and installed Mojave 10.14.3.
Now Bluetooth can't be recognized, can not apply CAT.

Any ideas how to get it working? I'm super new to all this, any help is appreciated. Thanks

![screen shot 2019-03-07 at 2 09 32 am](https://user-images.githubusercontent.com/48293745/53903232-27d77780-407e-11e9-9027-1749aae85224.png)

![screen shot 2019-03-07 at 2 06 27 am](https://user-images.githubusercontent.com/48293745/53903060-a8e23f00-407d-11e9-8129-3631023493de.png)

**Comment by codethought** [comment]
I've given up trying to get CAT to work with Mojave..  When there's an update it stops working and I have to start over again completely..  Even then there are problems if a reboot with it remaining stable.

I think High Sierra will be the last version of macOS I run on my cMP 5,1

**Comment by timetoseek** [comment]
> I've given up trying to get CAT to work with Mojave.. When there's an update it stops working and I have to start over again completely.. Even then there are problems if a reboot with it remaining stable.

@codethought please don't give up just yet!  Here's a link to the steps I took to get it working on my 2012 Mac Pro:  https://github.com/dokterdok/Continuity-Activation-Tool/issues/450#issuecomment-453380542

 Good luck! 🍀

**Comment by ertik** [comment]
@tedteng85 
Try Instal macOS from USB - it don’t delete data, only recovery BT kexts and restore other hacks. Or restore original kexts from backUP.

@all
MBP 17“ Late 2011
BCM94331PCIEBT4CAX
BD ROM included ACCS
16 GB RAM
SamSung SSD 480 GB
macOS 10.13.6

AirDrop:
disable SIP
CAT use normally

HandOff and others:
method with patch of IO80211Family.kext and change boolean in plist

Plist can be edited by TextEdit -> copy original file on Desktop (or somewhere else), edit normally and copy to original location edited plist - authorize admin and all is OK.

RESTART
Enable SIP
BOOT

Complete working https://support.apple.com/en-us/HT204681

**Comment by pcolvin15** [comment]
Installed Mojave 10.14.3 onto MacBook Pro Late2011 (8,3) and hit a number of errors using the continuity tool, including inability to backup the kexts, and I lost use of my wifi card after running it (no hardware found). Continuity was working fine under 10.14.2.

Since I hadn't saved the kexts off manually first I ended up reinstalling and patching 10.14.3.  But afterward I thought best to look to see what kexts are being modified and to save those off manually. 

In looking through the code in contitool.sh I found two things that need to be worked through to clear the Mojave errors.  One is that the lack of access to the directory location for the backup/archive folders for the pre-modification/post-modification kexts.  An easy fix was to go edit the code and put in an unrestricted non-system pathway (such as ~/Downloads) into the variables for backupFolderBeforePatch and backupFolderAfterPatch.  The difficult one is that Mojave no longer has a kext named AirPortBrcm4360.kext.  The only AirPort kext is now named AirPortBrcmNIC-MFG.kext.  So the coding will not work.  This is the end of what I can do as a programmer as I don't know what should be patched in that file.

**Comment by toro0411** [comment]
I install Macair 4.1 OS 10.14.5 it's working.

**Comment by Okami3814** [comment]
Anyone know what wifi/bt card i need in order to get handoff working on a late macbook from 2009 (6,1) i managed to get airdrop to work with an external bt dongle that uses bt 4.0 but handoff is still an elusive feature.

**Comment by robog0bo** [comment]
Just a reminder to move the zip out of the downloads folder before extracting. Otherwise it might wind up sandboxed and cause all sorts of problems.

**Comment by atomizasser** [comment]
In the final version of Catalina 10.15 this method is not working. After replace board id in IO80211.family wifi is disable and i cannot turn it on.

**Comment by LuckyDime** [comment]
Hi all. I have a mac mini mid 2011, and I've installed dosdude's Catalina. In bluetooth settings, I've seen that it is support low energy mode but not handoff. So I couldn"t use sidecar with my iPad air 3.

Any idea?

**Comment by anQluck** [comment]
In dosdude1’s Catalina thread on Macrumours this problem already solved. All working including Handoff. Go get it there
https://forums.macrumors.com/threads/macos-10-15-catalina-on-unsupported-macs.2183772/page-1
